### DISCUSSION TITLE
x-data with function

### DISCUSSION THREAD

**Discussion by DriverCat** [alpinejs/alpine#2818] [1. Help] [answered]
I have an element, where I init x-data with function, both or them are in the same blade (Laravel):
```
<div x-data="app()"></div>

<script>
function app() {
    return ... all properties etc...
}
</script>
```
This workds fine, but when I move app() function from the blade to custom.js file and make mix (npm run dev), I get an error: 
```Alpine Expression Error: app is not defined```

Is it possible to have init function in outer .js file, like mix?

**Comment by SimoTod** [comment, marked as answer]
Yes, but the function needs to be available on the global scope. Mix by default doesn't expose anything on window so you need to manually assign it (somethings like `window.app = function() {...}`). If you use Alpine V3, you probably want to have a look at Alpine.data as well.

**Reply by DriverCat** [reply]
Used Alpine.data('app' ... in app.js bundle.
Works great, thank you for pointing me to the right docs place!
